daemons/fenced/fenced_commands.c

```diff
@@ -1164,6 +1170,8 @@ stonith_device_register(xmlNode * msg, const char **desc, gboolean from_cib)
     stonith_device_t *dup = NULL;
     stonith_device_t *device = build_device_from_xml(msg);
 
+    CRM_CHECK(device != NULL, return -ENOMEM);
+
```

Comment:
pacemaker-fenced doesn't currently check the return value of stonith_device_register() -- you'll have to either add those checks (with some sort of appropriate error handling), or CRM_ASSERT() here instead

Reply:
I know, but that's plain rabbit holing, isn't it?
That's the problem with wrongly maintained code at the origin.

Reply:
With the existing code, if device is NULL, it will segfault (which can be considered equivalent to an assertion). With this change, it will proceed as if it succeeded, and likely fail at some other point where the cause may not be as obvious. If we change the error handling of a function, we need to make sure the callers work with that -- that's not rabbit holing, that's completeness.

Reply:
Will look at that properly, but I am getting vertigos everytime such a deep
hole in error propagation pops up :-/

Reply:
Just put an assert here rather than an error return, that's how we generally handle memory errors (we use dynamic allocation so much that there's no point in continuing, and if memory is exhausted, the node is not likely to be functioning properly anyway).